x-pack/plugins/uptime/public/components/functional/snapshot_histogram.tsx

```diff
@@ -30,31 +29,27 @@ export const SnapshotHistogram = ({
   histogram,
   primaryColor,
   windowWidth,
-}: SnapshotHistogramProps) => {
-  const { upSeriesData, downSeriesData } = formatHistogramData(histogram);
-
-  return (
-    <EuiSeriesChart
-      width={windowWidth * windowRatio}
-      height={120}
-      stackBy="y"
-      xType={EuiSeriesChartUtils.SCALE.TIME}
-      xCrosshairFormat="YYYY-MM-DD hh:mmZ"
-    >
-      <EuiHistogramSeries
-        data={upSeriesData}
-        name={i18n.translate('xpack.uptime.snapshotHistogram.series.upLabel', {
-          defaultMessage: 'Up',
-        })}
-        color={primaryColor}
-      />
-      <EuiHistogramSeries
-        data={downSeriesData}
-        name={i18n.translate('xpack.uptime.snapshotHistogram.series.downLabel', {
-          defaultMessage: 'Down',
-        })}
-        color={dangerColor}
-      />
-    </EuiSeriesChart>
-  );
-};
+}: SnapshotHistogramProps) => (
+  <EuiSeriesChart
+    width={windowWidth * windowRatio}
+    height={120}
+    stackBy="y"
+    xType={EuiSeriesChartUtils.SCALE.TIME}
+    xCrosshairFormat="YYYY-MM-DD hh:mmZ"
+  >
+    <EuiHistogramSeries
+      data={histogram.map(({ x, x0, upCount }) => ({ x, x0, y: upCount }))}
+      name={i18n.translate('xpack.uptime.snapshotHistogram.series.upLabel', {
+        defaultMessage: 'Up',
+      })}
+      color={primaryColor}
+    />
+    <EuiHistogramSeries
+      data={histogram.map(({ x, x0, downCount }) => ({ x, x0, y: downCount }))}
```

Comment:
I like the compactness here.

Reply:
++ I've always felt there's too much custom formatting happening on the client. Our GQL API is built specifically for this UI, the data should be more-or-less pre-formatted before it arrives.